diff --git a/src/bpm/runc/client/client.go b/src/bpm/runc/client/client.go
index f9a2591c..e50a9390 100644
--- a/src/bpm/runc/client/client.go
+++ b/src/bpm/runc/client/client.go
@@ -228,6 +228,12 @@ func (c *RuncClient) DeleteContainer(containerID string) error {
 	return runcCmd.Run()
 }
 
+func (c *RuncClient) DeleteState(containerID string) error {
+	args := []string{"-rf", fmt.Sprintf("%s/%s", c.runcRoot, containerID)}
+	cmd := exec.Command("rm", args...)
+	return cmd.Run()
+}
+
 func (*RuncClient) DestroyBundle(bundlePath string) error {
 	return os.RemoveAll(bundlePath)
 }
diff --git a/src/bpm/runc/lifecycle/lifecycle.go b/src/bpm/runc/lifecycle/lifecycle.go
index 0621e2d0..cf25b158 100644
--- a/src/bpm/runc/lifecycle/lifecycle.go
+++ b/src/bpm/runc/lifecycle/lifecycle.go
@@ -74,6 +74,7 @@ type RuncClient interface {
 	ContainerState(containerID string) (*specs.State, error)
 	ListContainers() ([]client.ContainerState, error)
 	SignalContainer(containerID string, signal client.Signal) error
+	DeleteState(containerID string) error
 	DeleteContainer(containerID string) error
 	DestroyBundle(bundlePath string) error
 }
@@ -274,6 +275,11 @@ func (j *RuncLifecycle) StopProcess(logger lager.Logger, cfg *config.BPMConfig,
 }
 
 func (j *RuncLifecycle) RemoveProcess(logger lager.Logger, cfg *config.BPMConfig) error {
+	logger.Info("forcefully-deleting-runc-state")
+	if err := j.runcClient.DeleteState(cfg.ContainerID()); err != nil {
+		return err
+	}
+
 	logger.Info("forcefully-deleting-container")
 	if err := j.runcClient.DeleteContainer(cfg.ContainerID()); err != nil {
 		return err
